compiler/rustc_driver/src/pointcut.rs

```diff
@@ -0,0 +1,625 @@
+use rustc_data_structures::fx::FxHashMap;
```

Comment:
I don't think this logic should be implemented in `rustc_driver`. At the very least it should be pushed back to `rustc_interface` (otherwise rustdoc can't support it), but preferably it should be elsewhere.